common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt

```diff
@@ -231,8 +232,12 @@ class HTTPClient(
             "X-Observer-Mode-Enabled" to if (appConfig.finishTransactions) "false" else "true",
             "X-Nonce" to nonce
         )
-            .plus(authenticationHeaders)
-            .plus(eTagManager.getETagHeader(urlPath, refreshETag))
+            .apply {
+                putAll(authenticationHeaders)
+                if (supportsETag) {
+                    putAll(eTagManager.getETagHeader(urlPath, refreshETag))
+                }
```

Comment:
FWIW this is why I did this: https://github.com/RevenueCat/purchases-ios/pull/2278

That doesn't apply here, but I think it might still be useful to have in the future.

Reply:
Yeah... I think I will just leave it 👍

Reply:
+1, no harm in sending it

Reply:
wait, so did we end up not sending it?

Reply:
just saw the revert PR, sorry for the noise. I'll shut up now until I'm actually caught up